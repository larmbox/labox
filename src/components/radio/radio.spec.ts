import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { LRadio } from '~/index';

test('mount component', async () => {
  expect(LRadio).toBeTruthy();

  const component = mount(LRadio, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain(
    'form-control radio radio-md radio-primary'
  );
  expect(component.html()).not.toContain('form-field');
});

test('register css variables', async () => {
  mount(LRadio);
  expect(
    document.getElementById('lx-theme-provider-global')!.innerText
  ).toContain('--lx-radio');
});

test('inherit attributes', async () => {
  const component = mount(LRadio, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
