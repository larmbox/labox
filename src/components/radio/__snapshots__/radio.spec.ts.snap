// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"form-control radio radio-md radio-primary\\">
  <div class=\\"radio-inner\\">
    <div><input id=\\"id\\" type=\\"radio\\" />
      <div class=\\"radio-box\\">
        <div class=\\"radio-inner-box\\"></div>
      </div>
      <!--v-if-->
    </div>
    <div class=\\"radio-context\\">
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
</div>"
`;
